src/com/xilinx/rapidwright/rwroute/GlobalSignalRouting.java

```diff
@@ -190,39 +193,93 @@ private static Map<String, List<Node>> getListOfNodesFromRoutes(Device device, M
      *                      for same net as we're routing), or unavailable (preserved for other net).
      */
     public static void symmetricClkRouting(Net clk, Device device, Function<Node,NodeStatus> getNodeStatus) {
-        List<ClockRegion> clockRegions = getClockRegionsOfNet(clk);
-        ClockRegion centroid = findCentroid(clk, device);
-
-        List<ClockRegion> upClockRegions = new ArrayList<>();
-        List<ClockRegion> downClockRegions = new ArrayList<>();
-        // divides clock regions into two groups
-        divideClockRegions(clockRegions, centroid, upClockRegions, downClockRegions);
-
-        RouteNode clkRoutingLine = UltraScaleClockRouting.routeBUFGToNearestRoutingTrack(clk);// first HROUTE
-        RouteNode centroidHRouteNode = UltraScaleClockRouting.routeToCentroid(clk, clkRoutingLine, centroid, true, true);
-
-        RouteNode vrouteUp = null;
-        RouteNode vrouteDown;
-        // Two VROUTEs going up and down
-        ClockRegion aboveCentroid = upClockRegions.isEmpty() ? null : centroid.getNeighborClockRegion(1, 0);
-        if (aboveCentroid != null) {
-            vrouteUp = UltraScaleClockRouting.routeToCentroid(clk, centroidHRouteNode, aboveCentroid, true, false);
-        }
-        vrouteDown = UltraScaleClockRouting.routeToCentroid(clk, centroidHRouteNode, centroid.getNeighborClockRegion(0, 0), true, false);
+        if (device.getSeries() == Series.UltraScale || device.getSeries() == Series.UltraScalePlus) {
```

Comment:
Would it make sense to refactor the bodies of the if/else if into their own method inside `UltraScaleClockRouting.java` and `VersalClockRouting.java`?